src/neural/cuda/inputs_outputs.h

```diff
@@ -134,6 +136,8 @@ struct InputsOutputs {
 
   // cublas handle used to run the network
   cublasHandle_t cublas_;
+
+  bool has_moves_left_ = false;
```

Comment:
Wouldn't it be possible to just check op_moves_left_mem_ to be nullptr? (and initialize it like that)

Reply:
I did not find a very good way to test if the `op_moves_left_mem_` has been allocated. `op_moves_left_mem_ != nullptr` doesn't work as even the unallocated one still gets a pointer value. The only thing I have seen is that `op_moves_left_mem_ == op_moves_left_mem_gpu_` for cases where it has been allocated (and would be different if not allocated). Though I'm not sure if that is a safe or good way to go about it.

Reply:
I didn't understand the comment initially. Fixed now. Initialized to `nullptr` and then used as the check for deallocation.